File: 179-largest-number/largest-number.py
class Solution(object):
    def largestNumber(self, nums):
        if(len(nums)==1):
            return str(nums[0])
        elif(sum(nums)==0):
            return "0"
        arr = list(map(str, nums))
        for i in range(len(nums)):
            for j in range(1,len(nums)):
                if(arr[j-1]+arr[j]<arr[j]+arr[j-1]):
                    arr[j-1],arr[j]=arr[j],arr[j-1]
        return ''.join(arr)



        
        

File: 179-largest-number/largest-number.java
import java.util.*;

public class Solution {
    public String largestNumber(int[] nums) {
       ArrayList<String> arr = new ArrayList<>();
       for(int i=0;i<nums.length;i++)
       {
         arr.add(Integer.toString(nums[i]));
       }
       arr.sort((a,b)->{
          if((a+b).compareTo(b+a)>0) return -1;
          return 1;
       });
       String s = "";
       for(int i=0;i<arr.size();i++) 
       {
          s+=arr.get(i);
       }
       if(s.charAt(0)=='0') return "0";
       return s;
    }
}
